bundles/auth/org.eclipse.smarthome.auth.oauth2client.test/src/main/java/org/eclipse/smarthome/auth/oauth2client/test/TestAgent.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.auth.oauth2client.test;
+
+import java.io.IOException;
+
+import org.eclipse.smarthome.core.auth.client.oauth2.AccessTokenResponse;
+import org.eclipse.smarthome.core.auth.client.oauth2.OAuthClientService;
+import org.eclipse.smarthome.core.auth.client.oauth2.OAuthException;
+import org.eclipse.smarthome.core.auth.client.oauth2.OAuthResponseException;
+
+/**
+ * For testing
+ *
+ * @author Gary Tse - initial contribution
+ *
+ */
+public interface TestAgent {
```

Comment:
It is used by ConsoleOAuthCommandExtension.

Reply:
You can have an interface but it is not necessary. You can expose the implementations as `@Component(service = ResourceOwnerTestAgent.class)` and `@Component(service = AuthorizationCodeTestAgent.class)`. Since the consumers reside in the same bundle, you don't need a service interface.

Reply:
Thx for suggestion.

Reply:
Yes, only the public methods will be accessible if you are accessing the DS Component from a different package, otherwise you can also access the package private methods if the consumer is in the same package as the DS Components.

Reply:
As discussed, leave it as it is